uat/mqtt-client-control/src/main/java/com/aws/greengrass/testing/mqtt/client/control/api/EventStorage.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.testing.mqtt.client.control.api;
+
+import lombok.NonNull;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * Interface to storage of the events.
+ */
+public interface EventStorage {
+
+    /**
+     * Removes all events from storage.
+     */
+    void clear();
+
+    /**
+     * Adds event to storage.
+     *
+     * @param event the event to add
+     */
+    void addEvent(@NonNull Event event);
+
+    /**
+     * Gets matched events from storage immediately.
+     *
+     * @param filter the filter of events
+     * @return the list of matched events
+     */
+    List<Event> getEvents(@NonNull EventFilter filter);
```

Comment:
I think that all this filtering should be done in our UAT itself and not in the client control. 

The agent should expose an API for callbacks on connect, disconnect, publish, subscribe and anything else that's needed.
The UAT will then listen for those events and can handle it however it wants.

Reply:
Yes, actually whole EventStorage is an add-on to Control.
Due to building blocks is not yet ready I place it to Control

Reply:
Let's get the work on the UAT then. This change seems really large for what it is trying to accomplish and I believe that doing this in the UAT will be much more reasonable.

Reply:
It not yet fully finished due to collect only MQTT messages. We are going to collect all events like agent registration/unregistration, MQTT connection/disconnection, and sure MQTT messages.

Ok, will rework to keep it separately from Control on UAT level.

Reply:
Moved to testing-features in 07f5113bf5eb1e3c7c7f99ea87e915082be7d665